### PR TITLE
[Benchmark] Add benchmarks for TF Training

### PR DESCRIPTION
**Pull request by patrickvonplaten**
This PR adds train functions to TF benchmarks including tests.
The notebook is updated accordingly as well.

### CODE REVIEW

File: src/transformers/benchmark/benchmark.py [hunk 1]
```diff
@@ -175,12 +175,12 @@ def _prepare_train_func(self, model_name: str, batch_size: int, sequence_length:
         def compute_loss_and_backprob_encoder():
             loss = train_model(input_ids, labels=input_ids)[0]
             loss.backward()
-            train_model.zero_grad()
```
Comment (patrickvonplaten) [inline src/transformers/benchmark/benchmark.py:178]: unnecessary call => remove

File: src/transformers/benchmark/benchmark.py [hunk 2]
```diff
@@ -157,7 +157,7 @@ def _prepare_train_func(self, model_name: str, batch_size: int, sequence_length:
         else:
             train_model = model
 
-        model.eval()
+        model.train()
```
Comment (patrickvonplaten) [inline src/transformers/benchmark/benchmark.py:160]: This disabled dropout before.

**Comment by patrickvonplaten** [comment]
Merging. Pinging @LysandreJik for notification.
